src/ucx_van.h

```diff
@@ -640,6 +642,7 @@ class UCXVan : public Van {
     }
 
     ucp_tag_t tag       = MakeTag(my_node_.id, UCX_TAG_META);
+    tag                 = tag | (msg.meta.key & 0xFFFFFFFF);
```

Comment:
better update `MakeTag` in a way that it accepts key, like `MakeTag(my_node_.id, UCX_TAG_META, key)`
and do all  tag manipulations inside `MakeTag`